vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -373,9 +348,23 @@ void learn_or_predict(ccb& data, multi_learner& base, multi_ex& examples)
   // split shared, actions and slots
   if (!split_multi_example_and_stash_labels(examples, data)) { return; }
 
-#ifndef NDEBUG
-  if (!sanity_checks<is_learn>(data)) { return; }
-#endif
+  if (data.slots.size() > data.actions.size())
+  {
+    std::stringstream msg;
+    msg << "ccb_adf_explore: badly formatted example - number of actions " << data.actions.size()
+        << " must be greater than the number of slots " << data.slots.size();
+    THROW(msg.str())
```

Comment:
is throwing the right way to go here since we are trying to reduce throw's in vw? Should we just cleanup (if there is anything to cleanup) and return?

Reply:
The problem is that whoever is calling this expects predictions etc to be filled out and by just returning that is not the case. This was an issue reported by a user that they were hitting in the wild. Error code would be better so we aren't adding a throw here but we don't have the infra in place to do it yet.

Reply:
The problem was only in debug mode, which means that we are were doing something different in release mode

Reply:
They would crash in release mode AFAIK so the old sanity check just did nothing to help them